doc/blog/Wasm-JS series/Blog - WASM.md

```diff
@@ -0,0 +1,672 @@
+   {
+       (dependencyobject as PrismJsView)?.UpdateDisplay();
+   }
+   
+   private static void LanguageChanged(DependencyObject dependencyobject, DependencyPropertyChangedEventArgs args)
+   {
+       (dependencyobject as PrismJsView)?.UpdateDisplay(args.OldValue as string, args.NewValue as string);
+   }
+   ```
+
+4. We also need to update the result when the control is loaded in the DOM. So we need to change the constructor again like this:
+
+   ``` csharp
+   public PrismJsView() : base("code") // PrismJS requires a <code> element
+   {
+       // This is required to set to <code> style for PrismJS to works well
+       // https://github.com/PrismJS/prism/issues/1237#issuecomment-369846817
+       SetStyle("white-space", "pre-wrap");
+   
+       // Update the display when the element is loaded in the DOM
+       Loaded += (snd, evt) => UpdateDisplay(newLanguage: Language);
+   }
+   ```
+
+5. Compile & run.  It should work like this:
+   ![image-20200415135422628](image-20200415135422628.png)
+
+## 🔬 Going further
+
+That's a very simple integration because there's no _callback_ from HTML to managed code and _PrismJS_ is a self-contained framework.
+
+Few improvements could be done to make the code more _production ready_:
+
+* **Make the control multi-platform**. The simple way would be to use a WebView on other platforms: that way the exact same text-rendering framework would be used everywhere. Right now this code won't compile on other targets.
+* **Create script files instead of generating dynamic javascript**. That would have the advantage of improving performance and increase the ability to debug it. Few projects are also using TypeScript to generate javascript. This approach is done by Uno itself for the `Uno.UI.Wasm` project: https://github.com/unoplatform/uno/tree/master/src/Uno.UI.Wasm.
+* **Support more PrismJS features**. There's many [_plugins_ to PrismJS](https://prismjs.com/#plugins) you can play with. Most of them are very easy to implement.
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+# Article 2 - Callback to app from JavaScript
+
+In the previous article, a simple _syntax highlighter_ were used to enhance the display of text in HTML. But it's not enough for most apps: it's often required for JavaScript components to _call back_ to the application. The easiest way to do that in Uno-Wasm is by using [_DOM Events_](https://developer.mozilla.org/docs/Web/Guide/Events/Creating_and_triggering_events). Let's create an application illustrating how to use them. Application using Uno can [consume DOM Events and CustomEvents](https://platform.uno/docs/articles/wasm-custom-events.html) very easily.
+
+# Sample 2 - Integration of Flatpickr
+
+📝 [Flatpickr](https://flatpickr.js.org/) is a lightweight, self-contained _Date & Time Picker_. It's an easy way to explore how a JavaScript can call-back to _managed application code_. In this case, it will be used to report when the picker is opened and the picked date & time.
```

Comment:
```suggestion
📝 [Flatpickr](https://flatpickr.js.org/) is a lightweight, self-contained date and time picker. It's an easy way to explore how a JavaScript can call back to the managed application code. In this case, this will be used to report when the picker is opened and a date and time was picked.
```